src/components/MagicCodeInput.js

```diff
@@ -124,9 +125,12 @@ function MagicCodeInput(props) {
 
     const validateAndSubmit = () => {
         const numbers = decomposeString(props.value, props.maxLength);
-        if (!props.shouldSubmitOnComplete || _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length !== props.maxLength || props.network.isOffline) {
+        if (wasSubmitted || !props.shouldSubmitOnComplete || _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length !== props.maxLength || props.network.isOffline) {
             return;
         }
+        if (!wasSubmitted) {
```

Comment:
At this point, `wasSubmitted` is always `false`

Reply:
As this is a minor thing (as it might be argued that the explicit `if` adds some readability), I'll still approve. But I would personally remove the `if`.